src/plugins/kibana_react/public/field_button/field_button.tsx

```diff
@@ -0,0 +1,118 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import './field_button.scss';
+import classNames from 'classnames';
+import React, { ReactNode, HTMLAttributes, ButtonHTMLAttributes } from 'react';
+import { CommonProps } from '@elastic/eui';
+
+export interface FieldButtonProps extends HTMLAttributes<HTMLDivElement> {
+  /**
+   * Label for the button
+   */
+  fieldName: ReactNode;
+  /**
+   * Icon representing the field type.
+   * Recommend using FieldIcon
+   */
+  fieldIcon?: ReactNode;
+  /**
+   * An optional node to place inside and at the end of the <button>
+   */
+  fieldInfoIcon?: ReactNode;
+  /**
+   * An optional node to place outside of and to the right of the <button>
+   */
+  fieldAction?: ReactNode;
+  /**
+   * Adds a forced focus ring to the whole component
+   */
+  isActive?: boolean;
+  /**
+   * Styles the component differently to indicate it is draggable
+   */
+  isDraggable?: boolean;
+  /**
+   * Use the small size in condensed areas
+   */
+  size?: ButtonSize;
+  className?: string;
+  /**
+   * The component always renders a `<button>` and therefore will always need an `onClick`
+   */
+  onClick: () => void;
+  /**
+   * Pass more button props to the actual `<button>` element
+   */
+  buttonProps?: ButtonHTMLAttributes<HTMLButtonElement> & CommonProps;
+}
+
+/**
```

Comment:
This seems more complicated than necessary to me - also `m` and `s` are still defined in two places (the class name map and the type). This simplifies it:
```ts
const sizeToClassNameMap = {
  s: 'kbnFieldButton--small',
  m: null,
} as const;

export type ButtonSize = keyof typeof sizeToClassNameMap;

export const SIZES = Object.keys(sizeToClassNameMap) as ButtonSize[];
```

Reply:
Can you be more explicit about what you find complicated?

The reason for adding `buttonProps` is to make it easy for devs who want to be able to pass along something like a `data-test-subj` or an `aria-label` directly to this **inner** button. It's the easiest way to support these extra props.

Reply:
This is not about the button props but the lines below ( https://github.com/elastic/kibana/pull/73226/files/af06e49d49aa37b1fd09e4d448bbd168179d9560#diff-a747347502de0443374f2c31281b461fR67-R80 )

This is just a nit, with my proposal the code will do the same thing but be shorter with less redundancy